Revert upstream commit f5119db

--- origsrc/gnome-online-accounts-3.10.2/configure.ac	2013-11-12 09:55:47.000000000 -0600
+++ src/gnome-online-accounts-3.10.2/configure.ac	2014-03-19 17:29:32.849806700 -0500
@@ -91,7 +91,7 @@ PKG_CHECK_MODULES(GTK, [gtk+-3.0 >= 3.5.
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(WEBKIT_GTK, [webkitgtk-3.0 >= 2.1.90])
+PKG_CHECK_MODULES(WEBKIT_GTK, [webkitgtk-3.0])
 AC_SUBST(WEBKIT_GTK_CFLAGS)
 AC_SUBST(WEBKIT_GTK_LIBS)
 
--- origsrc/gnome-online-accounts-3.10.2/src/goabackend/goaflickrprovider.c	2013-11-12 09:55:47.000000000 -0600
+++ src/gnome-online-accounts-3.10.2/src/goabackend/goaflickrprovider.c	2014-03-19 17:30:00.291791400 -0500
@@ -285,8 +285,8 @@ is_deny_node (GoaOAuthProvider *provider
   if (!WEBKIT_DOM_IS_HTML_ANCHOR_ELEMENT (node))
     goto out;
 
-  element = WEBKIT_DOM_ELEMENT (node);
-  id = webkit_dom_element_get_id (element);
+  element = WEBKIT_DOM_HTML_ELEMENT (node);
+  id = webkit_dom_html_element_get_id (element);
   if (g_strcmp0 (id, "auth-disallow") != 0)
     goto out;
 
--- origsrc/gnome-online-accounts-3.10.2/src/goabackend/goagoogleprovider.c	2013-11-12 09:55:47.000000000 -0600
+++ src/gnome-online-accounts-3.10.2/src/goabackend/goagoogleprovider.c	2014-03-19 17:30:50.134620600 -0500
@@ -284,7 +284,7 @@ is_identity_node (GoaOAuth2Provider *pro
   if (g_strcmp0 (element_type, "email") != 0)
     goto out;
 
-  id = webkit_dom_element_get_id (WEBKIT_DOM_ELEMENT (element));
+  id = webkit_dom_html_element_get_id (WEBKIT_DOM_HTML_ELEMENT (element));
   if (g_strcmp0 (id, "Email") != 0)
     goto out;
 
