--- origsrc/gnome-online-accounts-3.14.2/configure.ac	2015-01-04 23:00:31.227140200 -0600
+++ src/gnome-online-accounts-3.14.2/configure.ac	2015-01-04 22:29:27.774011700 -0600
@@ -90,7 +90,7 @@ PKG_CHECK_MODULES(GTK, [gtk+-3.0 >= 3.11
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(WEBKIT_GTK, [webkitgtk-3.0 >= 2.1.90])
+PKG_CHECK_MODULES(WEBKIT_GTK, [webkitgtk-3.0])
 AC_SUBST(WEBKIT_GTK_CFLAGS)
 AC_SUBST(WEBKIT_GTK_LIBS)
 
--- origsrc/gnome-online-accounts-3.14.2/src/goabackend/goaflickrprovider.c	2015-01-04 23:00:31.233140900 -0600
+++ src/gnome-online-accounts-3.14.2/src/goabackend/goaflickrprovider.c	2015-01-04 22:52:35.280202600 -0600
@@ -267,7 +267,7 @@ get_identity_sync (GoaOAuthProvider  *pr
 static gboolean
 is_deny_node (GoaOAuthProvider *provider, WebKitDOMNode *node)
 {
-  WebKitDOMElement *element;
+  WebKitDOMHTMLElement *element;
   gboolean ret;
   gchar *id;
 
@@ -277,8 +277,8 @@ is_deny_node (GoaOAuthProvider *provider
   if (!WEBKIT_DOM_IS_HTML_ANCHOR_ELEMENT (node))
     goto out;
 
-  element = WEBKIT_DOM_ELEMENT (node);
-  id = webkit_dom_element_get_id (element);
+  element = WEBKIT_DOM_HTML_ELEMENT (node);
+  id = webkit_dom_html_element_get_id (element);
   if (g_strcmp0 (id, "auth-disallow") != 0)
     goto out;
 
--- origsrc/gnome-online-accounts-3.14.2/src/goabackend/goagoogleprovider.c	2015-01-04 23:00:31.240641900 -0600
+++ src/gnome-online-accounts-3.14.2/src/goabackend/goagoogleprovider.c	2015-01-04 22:29:27.804015500 -0600
@@ -290,7 +290,7 @@ is_identity_node (GoaOAuth2Provider *pro
   if (g_strcmp0 (element_type, "email") != 0)
     goto out;
 
-  id = webkit_dom_element_get_id (WEBKIT_DOM_ELEMENT (element));
+  id = webkit_dom_html_element_get_id (WEBKIT_DOM_HTML_ELEMENT (element));
   if (g_strcmp0 (id, "Email") != 0)
     goto out;
 
--- origsrc/gnome-online-accounts-3.14.2/src/goabackend/goapocketprovider.c	2014-11-12 08:14:32.000000000 -0600
+++ src/gnome-online-accounts-3.14.2/src/goabackend/goapocketprovider.c	2015-01-04 22:53:46.223211200 -0600
@@ -270,7 +270,7 @@ get_identity_sync (GoaOAuth2Provider  *p
 static gboolean
 is_deny_node (GoaOAuth2Provider *provider, WebKitDOMNode *node)
 {
-  WebKitDOMElement *element;
+  WebKitDOMHTMLElement *element;
   gboolean ret;
   gchar *id;
   gchar *class;
@@ -281,13 +281,13 @@ is_deny_node (GoaOAuth2Provider *provide
   text = NULL;
   ret = FALSE;
 
-  if (!WEBKIT_DOM_IS_ELEMENT (node))
+  if (!WEBKIT_DOM_IS_HTML_ELEMENT (node))
     goto out;
 
-  element = WEBKIT_DOM_ELEMENT (node);
+  element = WEBKIT_DOM_HTML_ELEMENT (node);
 
   /* Desktop version */
-  id = webkit_dom_element_get_id (element);
+  id = webkit_dom_html_element_get_id (element);
   if (g_strcmp0 (id, "denyButton") == 0)
     {
       ret = TRUE;
@@ -295,7 +295,7 @@ is_deny_node (GoaOAuth2Provider *provide
     }
 
   /* Mobile version */
-  class = webkit_dom_element_get_class_name (element);
+  class = webkit_dom_html_element_get_class_name (element);
   if (g_strcmp0 (class, "toolbarButton") != 0)
     goto out;
 
