--- origsrc/gnome-online-accounts-3.2.1/src/goa/Makefile.am	2011-09-02 21:31:12.000000000 -0500
+++ src/gnome-online-accounts-3.2.1/src/goa/Makefile.am	2011-11-17 19:07:34.911280700 -0600
@@ -84,6 +84,8 @@ libgoa_1_0_la_CFLAGS =						\
 	$(GLIB_CFLAGS) 						\
 	$(NULL)
 
+libgoa_1_0_la_LDFLAGS = -no-undefined
+
 libgoa_1_0_la_LIBADD = 						\
 	$(GLIB_LIBS) 						\
 	$(NULL)
--- origsrc/gnome-online-accounts-3.2.1/src/goabackend/Makefile.am	2011-09-27 20:07:47.000000000 -0500
+++ src/gnome-online-accounts-3.2.1/src/goabackend/Makefile.am	2011-11-17 19:56:24.632851200 -0600
@@ -89,6 +89,8 @@ libgoa_backend_1_0_la_CFLAGS =					\
 	$(LIBSOUP_CFLAGS)					\
 	$(NULL)
 
+libgoa_backend_1_0_la_LDFLAGS = -no-undefined
+
 libgoa_backend_1_0_la_LIBADD = 					\
 	$(top_builddir)/src/goa/libgoa-1.0.la			\
 	$(WEBKIT_GTK_LIBS)					\
